File: 0079-word-search/0079-word-search.cpp
class Solution {
public:
    bool wordsearch(vector<vector<char>>& board, const string& word, int i, int j, int k, int n, int m) {
        if (k >= word.length()) {
            return true;
        }
        if (i < 0 || j < 0 || i >= n || j >= m || board[i][j] != word[k]) {
            return false;
        }
        char temp = board[i][j]; // Store the current character
        board[i][j] = '#'; // Mark the cell as visited
        bool found = wordsearch(board, word, i + 1, j, k + 1, n, m) ||
                     wordsearch(board, word, i, j + 1, k + 1, n, m) ||
                     wordsearch(board, word, i - 1, j, k + 1, n, m) ||
                     wordsearch(board, word, i, j - 1, k + 1, n, m);
        board[i][j] = temp; // Restore the cell to its original state
        return found;
    }
    
    bool exist(vector<vector<char>>& board, const string& word) {
        int n = board.size(), m = board[0].size();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (board[i][j] == word[0] && wordsearch(board, word, i, j, 0, n, m)) {
                    return true;
                }
            }
        }
        return false;
    }
};

File: 0079-word-search/0079-word-search.js
/**
 * @param {character[][]} board
 * @param {string} word
 * @return {boolean}
 */

function helper(board, word, i,j,l){
    if(l >= word.length){
        return true;
    }
    if(i>=board.length || j>=board[0].length || i<0 || j<0 || board[i][j] != word[l]){
        return false;
    }
    const temp = board[i][j];
    board[i][j] = "#";
    let found = helper(board,word,i+1,j,l+1) || helper(board, word, i, j+1, l+1) || helper(board,word,i-1,j,l+1) || helper(board, word, i, j-1, l+1);
    board[i][j] = temp;
    return found;
}
var exist = function(board, word) {
    for(let i=0;i<board.length;i++){
        for(let j=0;j<board[0].length;j++){
            if (helper(board, word, i, j, 0)) {
                return true;
            }
        }
    }
    return false;
};